backend/test/core/usecase/AddTransactionToGroup.test.ts

```diff
@@ -73,71 +80,198 @@ describe("Add transaction to group (use case)", () => {
     test("Given a group composed of one user, when adding a transaction of -1€ from this user, an error 'Non positive transaction was submitted' should be returned", async () => {
       // Arrange
       const user = new GroupMember(1, "John");
-      const group = new Group(0, "group1", [user], []);
+      const secondUser = new GroupMember(2, "Dave");
+      const group = new Group(0, "group1", [user, secondUser], []);
       groupRepo.findById = vi.fn().mockResolvedValue(group);
       const addTransactionToGroup = new AddTransactionToGroup(groupRepo);
 
       // Act
-      const result = await addTransactionToGroup.execute(group.id, user.id, -1);
+      const result = await addTransactionToGroup.execute(
+        group.id,
+        user.id,
+        [secondUser.id],
+        -1
+      );
 
       // Assert
       expect(result).toStrictEqual({
         success: false,
         error: "Non positive transaction was submitted",
       });
     });
+
+    test("Given a group composed of one user, when adding a transaction from this user to this user, an error 'A GroupMember can not be exactly payer and the only recipient' should be returned", async () => {
+      // Arrange
+      const user = new GroupMember(1, "John");
+      const group = new Group(0, "group1", [user], []);
+      groupRepo.findById = vi.fn().mockResolvedValue(group);
+      const addTransactionToGroup = new AddTransactionToGroup(groupRepo);
+
+      // Act
+      const result = await addTransactionToGroup.execute(
+        group.id,
+        user.id,
+        [user.id],
+        1
+      );
+
+      // Assert
+      expect(result).toStrictEqual({
+        success: false,
+        error: "A GroupMember can not be exactly payer and the only recipient",
+      });
+    });
+
+    test("Given a group composed of one user, when adding a transaction from this user to another user, an error 'One recipient does not belong to the group or is duplicated' should be returned", async () => {
+      // Arrange
+      const user = new GroupMember(1, "John");
+      const otherUser = new GroupMember(2, "Dave");
+      const group = new Group(0, "group1", [user], []);
+      groupRepo.findById = vi.fn().mockResolvedValue(group);
+      const addTransactionToGroup = new AddTransactionToGroup(groupRepo);
+
+      // Act
+      const result = await addTransactionToGroup.execute(
+        group.id,
+        user.id,
+        [otherUser.id],
+        1
+      );
+
+      // Assert
+      expect(result).toStrictEqual({
+        success: false,
+        error: "One recipient does not belong to the group or is duplicated",
+      });
+    });
+
+    test("Given a group composed of two users, when adding a transaction from this user the other user two times, an error 'One recipient does not belong to the group or is duplicated' should be returned", async () => {
```

Comment:
From this user TO the other user ?

Reply:
This test isn't really clear, I suggest "Given a group composed of two users, when adding a transaction from the first user to twice the other user (recipientsId : [other user, other user]), an error 'One recipient does not belong to the group or is duplicated' should be returned"

Reply:
Agree, it was useful for the TDD, but now it tests the same behaviour